R/meta_validate.R

```diff
@@ -86,5 +88,17 @@ meta_validate <- function(meta) {
     }
   })
 
+  # check group factor level are the same
+  u_plan <- unique(meta$plan[, c("population", "observation")])
+  for (i in 1:nrow(u_plan)) {
+    key_pop <- u_plan[i, "population"]
+    key_obs <- u_plan[i, "observation"]
+    level_pop <- levels(meta$data_population[[metalite::collect_adam_mapping(meta, key_pop)$group]])
+    level_obs <- levels(meta$data_observation[[metalite::collect_adam_mapping(meta, key_obs)$group]])
+    if (!all(level_pop == level_obs)) {
```

Comment:
Is it possible the level values are the same but the orders are different? Meaning a set comparison instead of a exact match comparison. Or are they already sorted?